src/main/java/com/minecolonies/commands/ColonyInfo.java

```diff
@@ -0,0 +1,123 @@
+package com.minecolonies.commands;
+
+import com.minecolonies.colony.CitizenData;
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.colony.IColony;
+import com.minecolonies.colony.permissions.Permissions;
+import com.minecolonies.entity.EntityCitizen;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * List all colonies.
+ */
+public class ColonyInfo extends AbstractSingleCommand
+{
+
+    private static final String ID_TEXT                 = "§2ID: §f";
+    private static final String NAME_TEXT               = "§2 Name: §f";
+    private static final String MAYOR_TEXT              = "§2Mayor: §f";
+    private static final String COORDINATES_TEXT        = "§2Coordinates: §f";
+    private static final String COORDINATES_XYZ         = "§4x=§f%s §4y=§f%s §4z=§f%s";
+    private static final String CITIZENS                = "§2Citizens: §f";
+    private static final String NO_COLONY_FOUND_MESSAGE = "Colony %d not found.";
+
+    public static final String DESC                     = "info";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public ColonyInfo(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        int colonyId = -1;
+
+        //todo Find a way to get colony by name
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1135)